File: src/main/java/com/github/eendroroy/sdk/openweather/util/DigestUtil.java
package com.github.eendroroy.sdk.openweather.util;

import lombok.SneakyThrows;

import javax.xml.bind.DatatypeConverter;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

/**
 * @author indrajit
 */

public class DigestUtil {
    @SneakyThrows(NoSuchAlgorithmException.class)
    public String md5(String text){
        MessageDigest md = MessageDigest.getInstance("MD5");
        md.update(text.getBytes());
        byte[] digest = md.digest();
        return DatatypeConverter.printHexBinary(digest).toLowerCase();
    }
}

File: src/main/kotlin/com/github/eendroroy/sdk/openweathermap/util/DigestUtil.kt
package com.github.eendroroy.sdk.openweathermap.util

import java.security.MessageDigest
import java.security.NoSuchAlgorithmException
import javax.xml.bind.DatatypeConverter

/**
 * @author indrajit
 */
class DigestUtil {
    @Throws(NoSuchAlgorithmException::class)
    fun md5(text: String): String {
        val md = MessageDigest.getInstance("MD5")
        md.update(text.toByteArray())
        val digest = md.digest()
        return DatatypeConverter.printHexBinary(digest).toLowerCase()
    }
}